mc/client-s3.go

```diff
@@ -190,16 +191,92 @@ func (c *s3Client) ToogleLambdaNotificationStatus(enable bool, region, accountID
 	return nil
 }
 
+// Unwatch de-registers all bucket notification events for a given accountID.
+func (c *s3Client) Unwatch(params watchParams) *probe.Error {
+	// Extract bucket and object.
+	bucket, _ := c.url2BucketAndObject()
+	if err := isValidBucketName(bucket); err != nil {
+		return err
+	}
+
+	// Fetch the bucket location.
+	location, e := c.api.GetBucketLocation(bucket)
+	if e != nil {
+		return probe.NewError(e)
+	}
+
+	// Fetch any existing bucket notification on the bucket.
+	nConfig, e := c.api.GetBucketNotification(bucket)
+	if e != nil {
+		return probe.NewError(e)
+	}
+
+	// Construct account ARN.
+	accountARN := minio.NewArn("minio", "sns", location, params.accountID, "listen")
+
+	// Remote account ARN if any.
+	nConfig.RemoveTopicByArn(accountARN)
+
+	// Set back the new sets of notifications.
+	e = c.api.SetBucketNotification(bucket, nConfig)
+	if e != nil {
+		return probe.NewError(e)
+	}
+
+	// Success.
+	return nil
+}
+
+// Start watching on all bucket events for a given account ID.
 func (c *s3Client) Watch(params watchParams) (*watchObject, *probe.Error) {
 	eventChan := make(chan Event)
 	errorChan := make(chan *probe.Error)
 	doneChan := make(chan bool)
 
-	bucket, _ := c.url2BucketAndObject()
+	// Extract bucket and object.
+	bucket, object := c.url2BucketAndObject()
 	if err := isValidBucketName(bucket); err != nil {
 		return nil, err
 	}
 
+	// Fetch the bucket location.
+	location, e := c.api.GetBucketLocation(bucket)
+	if e != nil {
+		return nil, probe.NewError(e)
+	}
+
+	// Fetch any existing bucket notification on the bucket.
+	nConfig, e := c.api.GetBucketNotification(bucket)
+	if e != nil {
+		return nil, probe.NewError(e)
+	}
+
+	accountARN := minio.NewArn("minio", "sns", location, params.accountID, "listen")
+	// If there are no SNS topics configured, configure the first one.
+	shouldSetNotification := len(nConfig.TopicConfigs) == 0
+	if !shouldSetNotification {
+		// We found previously configure SNS topics, validate if current account-id is the same.
+		for _, topicConfig := range nConfig.TopicConfigs {
+			if topicConfig.Topic == accountARN.String() {
+				shouldSetNotification = false
+				break
+			}
+		}
+		shouldSetNotification = true
```

Comment:
maybe I'm looking not good, but it sets shouldSetNotification to false, breaks the for and immediately sets it to true again.